AutofacContrib.NSubstitute.Tests/ExampleFixture.cs

```diff
@@ -175,6 +184,54 @@ public void Example_test_with_substitute_for_concrete()
             Assert.That(result, Is.EqualTo(val3));
         }
 
+        [Test]
+        public void SubstituteForAndProvide()
+        {
+            const int val = 2;
+
+            using var utoSubstitute = AutoSubstitute.Configure()
+                .SubstituteFor<ConcreteClass>(val).Provide(out var concreteClass).Configured()

```

Comment:
I've been pondering what to call it and if we look at the verbs we already have the obvious choice is probably `Resolve`. Thoughts?

Side-note: this PR should add documentation to README.md to show how to use it.

Outside of that I think this one is good to go

Reply:
The challenge with Resolve is it implies it is available. This sets an `IProvidedValue<>` that will only be available once the build is called. Since there's hesitency on that method, I'll postpone it for now so that the other aspects can get merged.

Reply:
The `Provide(...)` method was the initial impetus for this change, but I realized while doing it I could plumb access to the container through the calls. This may arguably be a better way to do it.